pymeasure/display/widgets/sequencer_widget.py

```diff
@@ -193,75 +489,42 @@ def _add_tree_item(self, *, level=None, parameter=None, sequence=None):
             item is added. If level is 0, a root item will be added.
 
         :param parameter: If given, the parameter field is pre-filled
-        :param sequence: If given, the sequence field is pre-filled
         """
 
-        selected = self.tree.selectedItems()
+        selected = self.tree.selectionModel().selection().indexes()
 
         if len(selected) >= 1 and level != 0:
```

Comment:
Something goes wrong presently when adding an item using the "Add item" button: it always adds a level=0 item.
Pressing the button does not pass a level to this method (`_add_tree_item`), which therefore retains its default (0) value.

I think that you can omit the `and level != 0` (and even remove the 'Add root item' button and the `level` kwarg) because this way it will automatically add a root level item if no other item is selected. Otherwise, you could change the default value of level to None and retain the 'Add root item'.

Reply:
You are correct, this was a result of last-minute change where default value for level was set to 0 instead of `None`. I will revert this change.